src/test/acceptance/rest/test_rest_binary_search.py

```diff
@@ -0,0 +1,41 @@
+import json
+import time
+
+from storage.db_interface_backend import BackEndDbInterface
+from test.acceptance.base import TestAcceptanceBase
+from test.common_helper import get_firmware_for_rest_upload_test
+
+
+class TestRestBinarySearch(TestAcceptanceBase):
+
+    def setUp(self):
+        super().setUp()
+        self._start_backend()
+        self.db_backend = BackEndDbInterface(config=self.config)
+        time.sleep(1)  # wait for systems to start
+
+    def tearDown(self):
+        self.db_backend.shutdown()
+        self._stop_backend()
+        super().tearDown()
+
+    def test_binary_search(self):
+        self._upload_firmware()
+        time.sleep(2)  # wait for binary to be saved
+
+        data = {'rule_file': 'rule rulename {strings: $a = "MyTestRule" condition: $a }'}
+        rv = self.test_client.post('/rest/binary_search', data=json.dumps(data), follow_redirects=True)
+        result = json.loads(rv.data)
```

Comment:
jenkins fails on this line since `data` has type `bytes` while `json.loads` expects a `str`. Suspiciously this does not fail on my local machine. Nevertheless, using `.decode()` here makes sense.

Reply:
Also in line 33